### PR TITLE
Bring up-to-date for Elm 0.18.

### PR DESCRIPTION
**Pull request by seagreen**

### CODE REVIEW

File: elm-package.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-    "src"
-  ],
-  "exposed-modules": [],
-  "dependencies": {
-      "elm-lang/core": "4.0.1 <= v < 5.0.0",
-      "elm-lang/html": "1.1.0 <= v < 2.0.0",
-      "elm-lang/svg": "1.1.0 <= v < 2.0.0"
-  },
-  "elm-version": "0.17.0 <= v < 0.18.0"
+    "version": "1.0.0",
```
Comment (seagreen) [inline elm-package.json:2]: Normally I wouldn't put whitespaces in a PR, but they were added by the Elm tooling so it seemed best to just go with it.

File: src/HexGrid.elm [hunk 2]
```diff
@@ -417,10 +417,12 @@ line p1 p2 =
     []
   else
     let
-      n = toFloat <| distance p1 p2
-      pts = [0..n]
+      dist = distance p1 p2
```
Comment (seagreen) [inline src/HexGrid.elm:420]: One of the two non-mechanical changes in the PR. We can't use plain `n :: Float` to create `pts` anymore, but I'm not sure what change caused this.

File: src/HexGrid.elm [hunk 3]
```diff
@@ -498,9 +500,9 @@ pointRound (x, z) =
   let
     y = -x - z
     (rx, ry, rz) = (round x, round y, round z)
-    dx = abs (rx - x)
-    dy = abs (ry - y)
-    dz = abs (rz - z)
+    dx = abs (toFloat rx - x)
```
Comment (seagreen) [inline src/HexGrid.elm:503]: The second and last non-mechanical change in the PR. The compiler will no longer allow `rx` to be an `Int` here, but I'm not sure what changed to cause that.

**Comment by seagreen** [comment]
🚨 First ever Elm code I've written. You might want to look it over carefully=) 🚨 

I tried to make changes following the [Upgrading to 0.18](https://github.com/elm-lang/elm-platform/blob/master/upgrade-docs/0.18.md) guide exactly. I did run into two issues where the compiler would no longer allow variables to be `Int` that weren't explained by the 0.18 changes, I noted both in comments.

**Comment by danneu** [comment]
Thanks, I'll check it out.

**Comment by danneu** [comment]
Hey, I had to make a lot more changes to get it 0.18-ready.

I was going to merge your changes and then do the rest but I totally forgot. Sorry.

Diff: https://github.com/danneu/elm-hex-grid/commit/558fa7ecc069b9ed0b663206dfcbef3d20926e52

I also broke the demo into individual models to improve the performance.

**Comment by danneu** [comment]
@seagreen For some reason I thought I was looking at the Files Changed tab and thought the 3 lines you commented on were the only lines you changed. You can tell I'm not used to getting contributors on Github. :rofl: 

Thanks for the contribution!

**Comment by seagreen** [comment]
No worries, I dashed it off really quick anyway. Did you ever figure out what required the addition of the `toFloat` calls between 0.17 and 0.18?

**Comment by danneu** [comment]
@seagreen at a glance, it seems like Elm just fixed a bug.

```diff
@@ -498,9 +500,9 @@ pointRound (x, z) =
   let
     y = -x - z
     (rx, ry, rz) = (round x, round y, round z)
-    dx = abs (rx - x)
-    dy = abs (ry - y)
-    dz = abs (rz - z)
+    dx = abs (toFloat rx - x)
+    dy = abs (toFloat ry - y)
+    dz = abs (toFloat rz - z)
```

- `{x,y,z}` are floats (`{x,z}` from type signature, `y` from subtraction). 
- `round`ing each one produces integers `rx`, `ry`, `rz`. 

so it makes that `rx - x` i.e. `int - float` must be lifted back into `float - float`.

**Pull request merged**